api/osrd_infra/models/rolling_stock.py

```diff
@@ -26,11 +27,16 @@ class RollingStock(models.Model):
         help_text=_("A group of curves mapping speed (in m/s) to maximum traction (in newtons)"),
         validators=[PydanticValidator(EffortCurves)],
     )
-    power_class = models.CharField(
+    base_power_class = models.CharField(
         max_length=255,
         help_text=_("The power usage class of the train (optional because it is specific to SNCF)"),
         null=True,
     )
+    power_restrictions = models.JSONField(
+        help_text=_("Mapping from train's power restriction codes to power classes"),
```

Comment:
what's the difference between power restriction codes and power classes?

Reply:
A restriction code can be set on an area of an infrastructure.
The train driver then uses a notch that matches the restriction code.
The notch chosen affects the speed-effort curves and changes the power class of the train.

In our data sources, the notches and restrictions are merged.